source/fundamentals/builders/aggregation-expression-operations.txt

```diff
@@ -0,0 +1,1198 @@
+
+   * - `slice() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#slice(int,int)>`__
+     - :manual:`$slice </reference/operator/aggregation/slice/>`
+
+   * - `sum() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#sum(java.util.function.Function)>`__
+     - :manual:`$sum </reference/operator/aggregation/sum/>`
+
+   * - `union() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#union(com.mongodb.client.model.mql.MqlArray)>`__
+     - :manual:`$setUnion </reference/operator/aggregation/setUnion/>`
+
+   * - `unionArrays() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlArray.html#unionArrays(java.util.function.Function)>`__
+     - :manual:`$setUnion </reference/operator/aggregation/setUnion/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+.. tip::
+
+   When retrieving arrays, you must specify the type of array that is returned
+   if you want to continue working with the return value.
+
+Suppose you have a collection of movies, each of which contains an array
+of nested documents for upcoming showtimes. Each nested document
+contains an array that represents the total number of seats in the
+theater, where the first array entry is the number of premium seats and
+the second entry is the number of regular seats. Each nested document
+also contains the number of tickets that have already been bought for
+the showtime. A document in this collection might resemble the
+following:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "_id": ...,
+     "movie": "Hamlet",
+     "showtimes": [
+       {
+         "date": "May 14, 2023, 12:00 PM",
+         "seats": [ 20, 80 ],
+         "ticketsBought": 100
+       },
+       {
+         "date": "May 20, 2023, 08:00 PM",
+         "seats": [ 10, 40 ],
+         "ticketsBought": 34
+       }]
+   }
+
+The ``filter()`` method displays only the results matching the provided
+predicate. In this case, the predicate uses ``sum()`` to calculate the
+total number of seats and compares that value to the number of ``ticketsBought``
+with ``lt()``. The ``project()`` method stores these filtered results as a new
+``availableShowtimes`` array.
+
+The following code shows the pipeline for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   asList(project(fields(
+            computed("availableShowtimes",
+               current()
+                  .<MqlDocument>getArray("showtimes")
+                  .filter(showtime -> {
+                        var totalSeats = showtime
+                              .<MqlInteger>getArray("seats")
+                              .sum(n -> n);
+                        var isAvailable = showtime
+                              .getInteger("ticketsBought")
+                              .lt(totalSeats);
+                        return isAvailable;
+                  })
+   ))));
```

Comment:
```
   asList(project(fields(
       computed("availableShowtimes", current()
           ...
   ))));
```
As a convention for formatting all Java examples:
1. Let's indent each level by 4 spaces (this one had 9, 3, 5...), to save space and for consistency.
2. If there are multiple stages, we should put `project` etc. on a new level. If there are not multiple, keep them on the same line (as I think we do everywhere).
3. However, even single fields ("computed") should go on separate lines, because things get too long otherwise?
4. Just as in `.filter(showtime -> showtime` [...], showtime would be on the same line as filter, current should be on the same line in `computed("availableShowtimes", current()` (and then a linebreak to chain off of current, or whatever initiates the expression).
5. (pending a prior comment) We should extract current fields into variables: `var abc = current().getSomething("abc")`.
6. I think that `asList(match(expr(current()` belong on one line, because this would correspond to `.match(c -> c` if stages used the same pattern (or, for that matter, `.filter(x -> x` even in this new API). But I'm a bit hesitant to suggest this one because things get a bit long, and mildly inconsistent with compute/fields.
7. Parenthesis should line up with the lines that introduce them. For example, above, there are 3 parens on the first line, but 4 on the last one. One of those should be moved to a new line directly before, to match the paren on `computed(`. Alternatively, if it really should be collapsed, it should collapse with the prior line (so, `}))`).